src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/ReflectTypeDescriptionProvider.cs

```diff
@@ -69,14 +69,23 @@ internal sealed partial class ReflectTypeDescriptionProvider : TypeDescriptionPr
 
         // These are attributes that, when we discover them on interfaces, we do
         // not merge them into the attribute set for a class.
-        private static readonly Type[] s_skipInterfaceAttributeList = new Type[]
+        private static readonly Type[] s_skipInterfaceAttributeList = InitializeSkipInterfaceAttributeList();
```

Comment:
Can we add `UnconditionalSuppressMessage` to this field directly? It seems to target all applicable constructs.

Reply:
Unfortunately that won't work. The warning is coming from a static constructor - which in this case is generated by the compiler. Currently the trimmer has no logic to try to map the warning from a static constructor back to the field (not even sure if it's possible in most cases).
First I simply added an explicit static `.cctor`, but Roslyn doesn't like it - probably due to `beforefieldinit` optimization which is really only possible when no explicit `.cctor` is specified. So the static method is a workaround to that.